torch/_dynamo/eval_frame.py

```diff
@@ -166,6 +166,31 @@ def enable_dynamic(enable: bool = True, export: bool = False):
         yield
 
 
+@contextlib.contextmanager
+def enable_functionalization(enable: bool, args):
+    if not enable:
+        yield args
+        return
+
+    memo: Dict[torch.Tensor, torch.Tensor] = {}
+
+    def to_fun(t):
+        if isinstance(t, torch.Tensor):
+            if t in memo:
+                return memo[t]
+            r = torch._to_functional_tensor(t, mirror_autograd_meta=True)
+            memo[t] = r
+            return r
+        else:
+            return t
+
+    try:
+        torch._enable_functionalization(reapply_views=True)
+        yield pytree.tree_map(to_fun, args)
+    finally:
+        torch._disable_functionalization()
```

Comment:
I think this will still cause problems, if your `Interpreter(graph).run()` call lower down raises an exception. You need that to be wrapped by the try/finally to make sure that functionalization is always properly turned off afterwards.

One way would be to change this context manager into a decorator that wraps your interpreter call. Or you can just not bother with the decorator here, and manually try/finally around the interpreter call.

Reply:
Good catch! I will just do plain try/finally.